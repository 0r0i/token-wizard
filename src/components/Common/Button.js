import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
