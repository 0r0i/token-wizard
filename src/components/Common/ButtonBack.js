import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
