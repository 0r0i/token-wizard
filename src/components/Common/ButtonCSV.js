import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
