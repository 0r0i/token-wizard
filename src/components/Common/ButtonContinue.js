import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
