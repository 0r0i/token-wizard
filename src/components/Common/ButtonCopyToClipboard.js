import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
