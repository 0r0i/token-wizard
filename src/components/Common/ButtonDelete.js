import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
