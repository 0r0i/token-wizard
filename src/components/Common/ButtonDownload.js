import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
