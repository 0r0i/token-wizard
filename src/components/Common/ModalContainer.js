import React from 'react'

export const ModalContainer = props => {
  if (props.showModal)
    return (
      <div className="sw-ModalWindowBackdrop">
        <div className="sw-ModalWindow">
          {props.title ? (
            <h2 className="sw-ModalWindow_Title">
              {props.title}{' '}
              {props.hideModal ? (
                <div className="sw-ModalWindow_CloseButton" onClick={() => props.hideModal()} />
              ) : null}
            </h2>
          ) : null}
          {props.description ? <p className="sw-ModalWindow_description">{props.description}</p> : null}
          {props.children}
        </div>
      </div>
    )
  return null
}
