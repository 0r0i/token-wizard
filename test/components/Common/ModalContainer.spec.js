import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import render from 'react-test-renderer'
import { ModalContainer } from '../../../src/components/Common/ModalContainer'
import { configure, mount } from 'enzyme'

configure({ adapter: new Adapter() })

describe(`ModalContainer`, () => {
  let hideTheModalWindow

  beforeEach(() => {
    hideTheModalWindow = jest.fn()
  })

  it(`should call onClick event`, () => {
    // Given
    const wrapper = mount(
      <ModalContainer showModal={true} title="Modal Title" hideModal={hideTheModalWindow} description="Some text..." />
    )

    // When
    wrapper.find('.sw-ModalWindow_CloseButton').simulate('click')

    // Then
    expect(hideTheModalWindow).toHaveBeenCalled()
    expect(hideTheModalWindow).toHaveBeenCalledTimes(1)
  })

  it(`should render ModalContainer component`, () => {
    // Given
    const component = render.create(
      <ModalContainer showModal={true} title="Modal Title" hideModal={hideTheModalWindow} description="Some text..." />
    )

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
