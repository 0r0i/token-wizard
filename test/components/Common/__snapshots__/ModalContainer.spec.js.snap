// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainer should render ModalContainer component 1`] = `
<div
  className="sw-ModalWindowBackdrop"
>
  <div
    className="sw-ModalWindow"
  >
    <h2
      className="sw-ModalWindow_Title"
    >
      Modal Title
       
      <div
        className="sw-ModalWindow_CloseButton"
        onClick={[Function]}
      />
    </h2>
    <p
      className="sw-ModalWindow_description"
    >
      Some text...
    </p>
  </div>
</div>
`;
